openwisp_radius/api/serializers.py

```diff
@@ -403,11 +403,17 @@ def validate_phone_number(self, phone_number):
         return phone_number
 
     def validate(self, data):
+        old_phone_number = self.user.phone_number
         self.user.phone_number = data['phone_number']
```

Comment:
I think here we should simply remove this. The model validation should be changed so that if the user is not active having an empty phone number won't generate the validation error.

We should however raise a validation error in this serializer if the phone number is missing from the serializer data when sms verification is enabled.

Reply:
I Think I should just improve this code and allow `PhoneToken` to handle the errors that belongs to him as it has already be done [here](https://github.com/openwisp/openwisp-radius/pull/183/commits/5e98ab84bc20b0dd0fd9996e4110eb0181723342#diff-4854b36face6ccda7d4d4cbe14d4eef0c25257dbf67e88f049173d80ba5954c6L809)